### PR TITLE
Temporary Brexit Middleware

### PR DESCRIPTION
**Pull request by lc512k**
Get brexit discount data to show on marketing promos across the site
There's a good chance Product will not need this after all, but getting it ready just in case.

@wheresrhys @matthew-andrews @ifyio 

cc @Purvi22Patel this is the change to get the brexit price in the marketing promo.

### CODE REVIEW

File: src/anon/middleware.js [hunk 1]
```diff
@@ -19,7 +20,32 @@ function anonymousMiddleware(req, res, next){
...
+	const headers = {
+		'Content-Classification': req.query['ft-content-classification'] || req.get('FT-Content-Classification') || 'CONDITIONAL_STANDARD',
+		'Country-Code': req.query['country-code'] || req.get('country-code') || 'GBR',
+	};
+
+	if (headers['Content-Classification'] === '-') {
+		headers['Content-Classification'] = 'CONDITIONAL_STANDARD';
+	}
+
+	console.log(headers);
```
Comment (ironsidevsquincy) [inline src/anon/middleware.js]: 💣

Comment (lc512k) [inline src/anon/middleware.js]: removed :)

File: src/anon/middleware.js [hunk 2]
```diff
@@ -19,7 +19,31 @@ function anonymousMiddleware(req, res, next){
...
+		'Content-Classification': req.query['ft-content-classification'] || req.get('FT-Content-Classification') || 'CONDITIONAL_STANDARD',
+		'Country-Code': req.query['country-code'] || req.get('country-code') || 'GBR',
+	};
+
+	if (headers['Content-Classification'] === '-') {
+		headers['Content-Classification'] = 'CONDITIONAL_STANDARD';
+	}
+
+	return fetch('https://barrier-app.memb.ft.com/memb/barrier/v1/barrier-data', { headers: headers, timeout: 2000 })
+		.then(res => res.json())
```
Comment (matthew-andrews) [inline src/anon/middleware.js]: doesn't appear to be any error handling for this request if it fails or get a bad error status

File: src/anon/middleware.js [hunk 3]
```diff
@@ -19,7 +19,31 @@ function anonymousMiddleware(req, res, next){
 		|| req.url;
 	res.locals.navigationModel = new NavigationModel(res.locals.flags, res.locals.anon.userIsAnonymous, currentUrl);
 	res.vary('FT-Anonymous-User');
-	next();
+
+	if (res.locals.flags.brexitDiscount && res.locals.flags.brexitDiscountType) {
+		getBarrierData(req).then(function (response) {
+			res.brexitOfferBasePrice = response.viewData.subscriptionOptions.STANDARD.price.weekly;
```
Comment (matthew-andrews) [inline src/anon/middleware.js]: have we checked the vary headers?

**Comment by lc512k** [comment]
Product don't need this anymore (thankfully) :)

**Pull request closed without merging**